File: applications/hierarchy/src/main/resources/hwc/hierarchy/SinkImpl.cpp
// (c) https://github.com/MontiCore/monticore
#include "SinkImpl.h"
#include <iostream>

namespace montithings {
namespace hierarchy {

SinkResult
SinkImpl::getInitialValues ()
{
  return {};
}

SinkResult
SinkImpl::compute (SinkInput input)
{
  if (input.getValue ())
    {
      log("Sink | in: " +  input.getValue ().value ());
    }
  else
    { std::cout << "No data." << std::endl; }
  return {};
}

}}

File: applications/mqtt-ports-with-cd/src/main/resources/hwc/hierarchy/SinkImpl.py
from SinkImplTOP import SinkImplTOP, SinkInput, SinkResult

class SinkImpl(SinkImplTOP):

    def getInitialValues(self) -> SinkResult:
        return SinkResult()

    def compute(self, _input: SinkInput) -> SinkResult:
        print(_input.payload)
        return SinkResult()